README.mdx

```diff
@@ -0,0 +1,198 @@
+project since we're building with attachments
+- If totally new to this, take a look at NonFungibleToken tutorials - TODO: Link these
+
+## Contract Overview
+
+- Contract interface implementation
+    - Events
+    - Nested resource definitions
+- Attribute overview
+- Note on `unsafeRandom()`
+- Opportunities to extend the contract
+
+## Deploy KittyVerse contract
+
+- Deploy NonFungibleToken to `0x01`
+- Deploy KittyVerse to `0x02`
+
+## Setup account & Mint KittyVerse NFT
+
+- send `tx_01_setup_kittyverse_account.cdc` signing with `0x02`
+- send `tx_02_mint_kittyverse_nft.cdc` signing with `0x02`
+- send `tx_02_mint_kittyverse_nft.cdc` again signing with `0x02`
+- query NFT.id - execute `script_01_get_kittyverse_ids.cdc` providing `0x02` as argument
+    - see two unique NFT IDS - take note of them, next let's prep them some hats
+
+# Define our Attachments - KittyHats
+
+## Attachments
+
+- What are they?
+    - In Cadence
+    - Simple mental model
+        - "Like glue between resources"
+        - ![KittyVerse & KittyHats attached](./diagrams/cat_with_hat.png)
+- Why do they matter?
+    - Can create attachments for anything without needing original developer's:
+        - Participation
+        - Permission
+
+### Overview
+
+- Conceptual
+    - Attachments are a special type that can be added to resources (doc link needed)
+    - You might be tempted to simply define all your functionality in the attachment, but you'll lose the existence and
+      uniqueness guarantees inherent to resources as well as make access control significantly harder for yourself
+        - As mentioned earlier, attachments are best thought of as glue between resources
+            - Define an attachment that can hold some resource, designed to be added to a base resource
+    - Note that you need access to the base resource to add an attachment to it
+        - So to add an attachment to an NFT, you need to first withdraw that NFT from its Collection
+- Syntax
+    - Accessing an attachment
+    ```cadence
+    let attachmentRef: &KittyHats.HatAttachment? = nft[KittyHats.HatAttachment]
+    ```
+    - Adding an attachment
+    ```cadence
+    access(self) fun addAttachment(toNFT: @KittyVerse.NFT): @KittyVerse.NFT {
+        // If attachment already exists, return
+        if toNFT[HatAttachment] != nil {
+            return <-toNFT
+        }
+        // Otherwise, add the attachment
+        return <- attach HatAttachment() to <- toNFT
+    }
+    // Otherwise, add the attachment
+    return <- attach HatAttachment() to <- toNFT
+    ```
+- Gotchas
+    - Accessing attachments of a given type requires a static type declaration
+    - Attempting to add an attachment to a resource when one already exists will result in a runtime error
+        - This is why we check if an attachment already exists before adding one in the example above
+        - You'll want to lean on simple helper methods to conditionally add attachments within your contracts
```

Comment:
Maybe leave Gotcha's to a later section where we summarize what was done in the tutorial and add that color then?

Reply:
Good idea. I can move it to the latter part for reflection on aspects of the design choices